runelite-client/src/main/java/net/runelite/client/plugins/deathindicator/DeathIndicatorPlugin.java

```diff
@@ -0,0 +1,267 @@
+package net.runelite.client.plugins.deathindicator;
+
+import com.google.common.eventbus.Subscribe;
+import com.google.inject.Provides;
+import java.awt.image.BufferedImage;
+import java.io.IOException;
+import java.time.temporal.ChronoUnit;
+import javax.imageio.ImageIO;
+import javax.inject.Inject;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.Actor;
+import net.runelite.api.Client;
+import net.runelite.api.NPC;
+import net.runelite.api.Player;
+import net.runelite.api.coords.WorldPoint;
+import net.runelite.api.events.ActorDeath;
+import net.runelite.api.events.ChatMessage;
+import net.runelite.api.events.ConfigChanged;
+import net.runelite.api.events.GameStateChanged;
+import net.runelite.api.events.GameTick;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.ui.overlay.infobox.InfoBoxManager;
+import net.runelite.client.ui.overlay.infobox.Timer;
+import net.runelite.client.ui.overlay.worldmap.WorldMapPointManager;
+
+@PluginDescriptor(
+		name = "Death Indicator"
+)
+@Slf4j
+public class DeathIndicatorPlugin extends Plugin
+{
+	public static final BufferedImage WORLDMAP_ARROW;
+	public static final BufferedImage BONES;
+
+	private boolean worldMapPointSet = false;
+	private boolean hasDied = false;
+	private Timer deathTimer = null;
+	private static int currentWorld = -1;
+	private static int deathWorld = -1;
+	private WorldPoint deathLocation = null;
+	private WorldPoint currentLocation = null;
+	private boolean deathMarkerReset = false;
+	private boolean playerHasRespawned = true;
+
+	@Inject
+	private Client client;
+
+	@Inject
+	private DeathIndicatorConfig config;
+
+	@Inject
+	private DeathIndicatorPlugin plugin;
```

Comment:
Use `this` instead of injecting a self reference

Reply:
reference to the DeathIndicatorPlugin by using the aforementioned `this` for example `deathTimer = new Timer(60, ChronoUnit.MINUTES, BONES, this);`

Reply:
Remove this line

Reply:
Oh okay, I got it, thanks guys I had no idea lol